utils_test.go

```diff
@@ -95,3 +98,50 @@ func TestVersionParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestExponentialBackoffValidCases(t *testing.T) {
+	testCases := []struct {
+		backoff            time.Duration
+		maxBackoff         time.Duration
+		retries            int
+		maxRetries         int
+		minBackoff         time.Duration
+		maxBackoffExpected time.Duration
+	}{
+		{100 * time.Millisecond, 2 * time.Second, 1, 5, 80 * time.Millisecond, 120 * time.Millisecond},
+		{100 * time.Millisecond, 2 * time.Second, 3, 5, 320 * time.Millisecond, 480 * time.Millisecond},
+		{100 * time.Millisecond, 2 * time.Second, 5, 5, 1280 * time.Millisecond, 1920 * time.Millisecond},
```

Comment:
The `backoff` and `maxBackoff` are now table-test entry invariant. Your original test design was a solid choice for testing of valid backoff and maxBackoff value.

Reply:
The current design is the same as the "original test design" (https://github.com/IBM/sarama/commit/d8df71458872d710cdda66fbec803b9ed91651fb) you mentioned. Is there any improvement that you recommend?

Reply:
No, the original original design did not include `backoff` and `maxBackoff` in the table. In fact, it made the `backoffFunc := NewExponentialBackoff(100 * time.Millisecond, 2 * time.Second)` at the top of the function before even listing out the test cases.